src/main/java/com/baidu/hugegraph/api/task/TaskAPI.java

```diff
@@ -68,12 +82,9 @@ public void delete(long id) {
 
     public void waitUntilTaskCompleted(long taskId, long seconds) {
         for (long pass = 0;; pass++) {
-            Task task = this.get(taskId);
-            if (task.success()) {
+            Task task = this.getFromCache(taskId);
+            if (this.completed(task)) {
```

Comment:
call this.removeFromCache(task.id()) here instead in completed()
and return task